doc/crypto/api/keys/management.rst

```diff
@@ -529,3 +536,456 @@ Key export
+
+    *   Set the object to all-bits-zero, for example:
+
+        .. code-block:: xref
+
+            psa_generate_key_iop_t operation;
+            memset(&operation, 0, sizeof(operation));
+
+    *   Initialize the object to logical zero values by declaring the object as static or global without an explicit initializer, for example:
+
+        .. code-block:: xref
+
+            static psa_generate_key_iop_t operation;
+
+    *   Initialize the object to the initializer `PSA_GENERATE_KEY_IOP_INIT`, for example:
+
+        .. code-block:: xref
+
+            psa_generate_key_iop_t operation = PSA_GENERATE_KEY_IOP_INIT;
+
+    *   Assign the result of the function `psa_generate_key_iop_init()` to the object, for example:
+
+        .. code-block:: xref
+
+            psa_generate_key_iop_t operation;
+            operation = psa_generate_key_iop_init();
+
+    This is an implementation-defined type.
+    Applications that make assumptions about the content of this object will result in implementation-specific behavior, and are non-portable.
+
+.. macro:: PSA_GENERATE_KEY_IOP_INIT
+    :definition: /* implementation-defined value */
+
+    .. summary::
+        This macro evaluates to an initializer for an interruptible key generation operation object of type `psa_generate_key_iop_t`.
+
+.. function:: psa_generate_key_iop_init
+
+    .. summary::
+        Return an initial value for an interruptible key generation operation object.
+
+    .. return:: psa_generate_key_iop_t
+
+.. function:: psa_generate_key_iop_get_num_ops
+
+    .. summary::
+        Get the number of *ops* that an interruptible key generation operation has taken so far.
+
+    .. param:: psa_generate_key_iop_t * operation
+        The interruptible key generation operation to inspect.
+
+    .. return:: uint32_t
+        Number of *ops* that the operation has taken so far.
+
+    After the interruptible operation has completed, the returned value is the number of *ops* required for the entire operation.
+    The value is reset to zero by a call to either `psa_generate_key_iop_setup()` or `psa_generate_key_iop_abort()`.
+
+    This function can be used to tune the value passed to `psa_iop_set_max_ops()`.
+
+    The value is undefined if the operation object has not been initialized.
+
+.. function:: psa_generate_key_iop_setup
+
+    .. summary::
+        Start an interruptible operation to generate a key or key pair.
+
+    .. param:: psa_generate_key_iop_t * operation
+        The interruptible key generation operation to set up.
+        It must have been initialized as per the documentation for `psa_generate_key_iop_t`, and be inactive.
+    .. param:: const psa_key_attributes_t * attributes
+        The attributes for the new key.
+
+        The following attributes are required for all keys:
+
+        *   The key type. It must not be an asymmetric public key.
```

Comment:
This suggests that implementations that support interruptible operations should support interruptible key generation for symmetric keys, even though operations _on_ symmetric keys don't have interruptible APIs. I think we should say that only key-pair types are expected to be supported, and implementations are expected to reject symmetric keys but they may accept them.

Reply:
Although on second thoughts, it's probably one less line of code to just systematically fall back to `psa_generate_key` for any key type that doesn't have interruptible support, and let it decide between public keys, key pairs with single-iop generation, and symmetric keys.

Reply:
It might be useful to state that an implementation is permitted to only support the use of this API for a subset of the key types that can be generated using `psa_generate_key()`?

Reply:
I've added a commit to note that implementations might only provide incremental generation for some key types.